### PR TITLE
[JENKINS-60866] Un-inline non-standard showDetails

### PR DESCRIPTION
**Pull request by Wadeck**
### AbstractModelObject
Un-inline the JavaScript / CSS from the AbstractModelObject/error.jelly by re-using the showDetails.
To trigger this page, I had to put a breakpoint in [Job.java#L1319](https://github.com/jenkinsci/jenkins/blob/1f946961b9416281fd8ec65eaa3304ed23838c25/core/src/main/java/hudson/model/Job.java#L1319), then configure a job, submit the form. The breakpoint is triggered, then, inside your favorite IDE, executing `json = new net.sf.json.JSONObject(true)` and let the rest of the code runs. This will throw a JSONException that will end display in that page. 
_I can bet there are other ways to trigger this error, but I prefered to keep focus on the un-inline as I had a way to trigger the page._

### ZFSInstaller
Un-inline the JavaScript / CSS from ZFSInstaller/askRootPassword.jelly by re-using the showDetails.
To trigger this page, I put a breakpoint in [ZFSInstaller.java#L237](https://github.com/jenkinsci/jenkins/blob/0e62b68012282648377cfac4fe6d01956d7c1f11/core/src/main/java/hudson/os/solaris/ZFSInstaller.java#L237), then browse to http://localhost:8080/jenkins/administrativeMonitor/hudson.os.solaris.ZFSInstaller/start, confirm the POST, then the breakpoint is triggered, you can execute: 
```
req.setAttribute("message", "testMessage");
rsp.forward(this,"askRootPassword",req);
```
and let the rest of the code run.

For both cases, the showDetails was using a custom inlined JavaScript while the showDetails feature was already present in hudson-behavior.js. The impact is that we are moving from a button to a link.

See [JENKINS-60866](https://issues.jenkins-ci.org/browse/JENKINS-60866).

<details>

<summary>Screenshots</summary>

### AbstractModelObject

Screenshot-2020-05-25_22-23-54

### ZFSInstaller

Screenshot-2020-05-25_22-44-00

</details>

### Proposed changelog entries

* (internal) Remove inline resources for non-standard showDetails

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: war/src/main/less/base/style.less [hunk 1]
```diff
@@ -2000,5 +2000,10 @@ body.no-sticker #bottom-sticker {
   height: 2em;
 }
 
+/* used by elements that are hidden by default but revealed throught JavaScript */
+.default-hidden {
```
Comment (Wadeck) [inline war/src/main/less/base/style.less:2004]: To not confuse with the new "hidden-by-default" used in PluginManager

File: core/src/main/resources/hudson/os/solaris/ZFSInstaller/askRootPassword.jelly [hunk 2]
```diff
@@ -45,18 +45,13 @@ THE SOFTWARE.
           </f:entry>
 
           <f:block>
-            <f:submit value="${%OK}" style="margin-top:1em;" />
+            <f:submit value="${%OK}" />
```
Comment (Wadeck) [inline core/src/main/resources/hudson/os/solaris/ZFSInstaller/askRootPassword.jelly:48]: The style was not applied in the final HTML generated

**Comment by Wadeck** [comment]
> [2020-05-25T20:56:03.024Z] write /var/lib/docker/tmp/GetImageBlob455469856: no space left on device

:'( 

Build restarted.

**Review by res0nance — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me, thanks for the patch!

We may merge it in 24 hours if no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information

**Pull request merged**